lib/rubocop/ast/node.rb

```diff
@@ -100,7 +100,8 @@ def complete?
       # part of it is changed.
       def updated(type = nil, children = nil, properties = {})
         properties[:location] ||= @location
-        self.class.new(type || @type, children || @children, properties)
+        klass = RuboCop::AST::Builder::NODE_MAP[type || @type] || self.class
```

Comment:
I guess the default class should not be the `self.class`, it should be the `RuboCop::AST::Node`.